modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -491,8 +559,17 @@ public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRu
                     charIndex = run.getOffsetAtX(x, trailing);
                     charIndex += textWidthPrevLine;
                     charIndex += relIndex;
+                    if (run.getLevel() % 2 != 0) {
```

Comment:
I wish there was an explanation of the meaning of `level`
And since there are several places where it checks for it being odd, I wish there was a method in TextRun with a descriptive name rather than this computation (and bit logic might be faster):

```
public boolean isLevelOdd() { // or whatever the meaning is
  return (level & 0x01) != 0; 
}
```

Reply:
Added comment and used bit logic in the condition.
Do you think we should create a method in TextRun? I believe it is out of scope of this PR as it will be used in other functions as well.

Reply:
Yes, creating such a new method in TextRun might be out of scope for this, unless we touch all the places where the bit logic is used.  Up to you, it's just a suggestion.

Reply:
I would prefer not to make this change in this PR. So I will keep this as it is for now.